pkg/storegateway/gateway.go

```diff
@@ -370,6 +372,10 @@ func (g *StoreGateway) waitRingStability(ctx context.Context, reason string) {
 	minWaiting := g.gatewayCfg.ShardingRing.WaitStabilityMinDuration
 	maxWaiting := g.gatewayCfg.ShardingRing.WaitStabilityMaxDuration
 
+	if !g.gatewayCfg.ShardingEnabled || minWaiting <= 0 {
```

Comment:
Isn't it duplicate of `shouldWaitRingStability`? I think we can keep one

Reply:
oops no that's supposed to be removed, I'll delete it